docs/rfcs/1011-non-blocking-invite-transport.md

```diff
@@ -0,0 +1,1238 @@
+            {
+                // The submitted step is too advanced
+                // Every step before must have been completed by both peers
+                "status": "step_too_advanced"
+            },
+            {
+                // The submitted step somehow changed during polling
+                "status": "step_mismatch"
+            }
+        ],
+        "nested_types": [
+            {
+                // GreeterStep should be identical to the one in invite_greeter_step.json5
+                "name": "GreeterStep",
+                "discriminant_field": "step",
+                "variants": [
+                    [...]  // This is described in the protocol update section
+                ]
+            },
+            {
+                // ClaimerStep should be identical to the one in invite_greeter_step.json5
+                "name": "ClaimerStep",
+                "discriminant_field": "step",
+                "variants": [
+                    [...]  // This is described in the protocol update section
+                ]
+            }
+        ]
+    }
+]
+```
+
+#### `invite_complete` command
+
+An extra authenticated command is added to complete an invitation.
+
+In the previous implementation, this was done automatically when the last exchange was performed. However, this is confusing in some cases:
+- Maybe the last exchange worked but the reply didn't reach one of the peers. In this case, the peer cannot retry the exchange since the invite is now completed.
+- In a shared recovery (shamir) invitation, the invitation must not complete after a successful attempt since there might be more exchanges to complete with other recipients.
+
+A better approach is then to explicitly complete the invitation with a dedicated command. The allowed authors for this command depends on the invitation type:
+- User invitation: any administrator or the freshly registered user are allowed
+- Device invitation: only the corresponding user is allowed
+- Shamir invitation: any recipient of the freshly recovered user are allowed
+
+Pseudo-code:
+
+```python
+def invite_complete(token: InvitationToken) ->
+  InvitationNotFound | InvitationCompleted | InvitationCancelled | Ok:
+    # Perform checks
+    [...]
+    # Call `complete_invitation`
+    [...]
+    # Return `Ok`
+    [...]
+```
+
+Schema definition:
+
+```json5
+[
+    {
+        "major_versions": [
+            4
+        ],
+        "req": {
+            "cmd": "invite_complete",
+            "fields": [
+                {
+                    "name": "token",
+                    "type": "InvitationToken"
+                }
+            ]
+        },
+        "reps": [
+            {
+                "status": "ok"
+            },
+            {
+                "status": "invitation_not_found"
+            },
+            {
+                "status": "invitation_already_completed"
+            },
+            {
+                "status": "invitation_cancelled"
+            }
```

Comment:
There should also be a `author_not_allowed` status

Reply:
d36e8a3440bf5a5cb645e168de63108650bc36bf